### PR TITLE
tests: using google storage when downloading ubuntu cloud images from gce

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is implemented because the preseed test is downloading this images
on each travis run. The idea is to make it faster and save networking by
using the google storage when running tests on gce.

### CODE REVIEW

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -49,25 +49,54 @@ get_qemu_for_nested_vm(){
...
+            ;;
+        *)
+            echo "unsupported system"
+            exit 1
+            ;;
+        esac
+}
+
+get_image_url_for_nested_vm(){
+    if [ "$SPREAD_BACKEND" = "google" ]; then
```
Comment (stolowski) [inline tests/lib/nested.sh]: Shouldn't this work for "google" and "google-nested"?

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -49,25 +49,54 @@ get_qemu_for_nested_vm(){
...
-    ubuntu-20.04-64)
-        echo "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img"
-        ;;
-    *)
-        echo "unsupported system"
-        exit 1
-        ;;
-    esac
+        ubuntu-16.04-64)
+            echo "https://storage.googleapis.com/spread-snapd-tests/images/xenial-server-cloudimg-amd64-disk1.img"
```
Comment (stolowski) [inline tests/lib/nested.sh]: What's the plan for refreshing these images in GC storage? Can we arrange to have them re-downloaded daily? It is important for preseeding tests to use relatively current images.

Comment (sergiocazzolato) [reply tests/lib/nested.sh]: No, this has to me manually downloaded, for that there is a cache system which I am going to integrate soon which allows that. 
Something that I can do is to create a spread cron which update those images monthly, does it work for you?

File: tests/lib/nested.sh (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/nested.sh]: thanks for the review, in fact not just google-nested but also google-sru and google-tpm

File: tests/lib/nested.sh (hunk 2 above)
Comment (stolowski) [reply tests/lib/nested.sh]: Yes, a cron job sounds good, however a month is too long, I think we should refresh them every few days or even every day.

Comment (sergiocazzolato) [reply tests/lib/nested.sh]: ok, I'll create the cron job in that case

Comment (stolowski) [reply tests/lib/nested.sh]: Actually it's only important for recent releases (19.10, 20.04). Old ones can be downloaded once a month.

**Pull request merged**